### ISSUE TITLE
Possibly bug/breaking change in parent items regular expression

### ISSUE THREAD

**Issue by achronis** [sdmx-twg/sdmx-ml#30]
Hello,

Just in case it is a bug, It seems that the item Parent restriction doesn't accept values starting with a number, e.g. CL_REF_AREA. 

https://github.com/sdmx-twg/sdmx-ml/blob/master/schemas/SDMXStructureCodelist.xsd#L77
https://github.com/sdmx-twg/sdmx-ml/blob/29f1a3d856c4259429f5ec0eae811653adc5cdb5/schemas/SDMXCommonReferences.xsd#L1597

```
Value '1' is not facet-valid with respect to pattern '[A-Za-z][A-Za-z0-9_\-]*(\.[A-Za-z][A-Za-z0-9_\-]*)*' for type 'SingleNCNameIDType'.line/column: 2/1846343cvc-type.3.1.3: The value '1' of element 'str:Parent'
```

But at the same time it is allowed to have a code that starts with a number
https://github.com/sdmx-twg/sdmx-ml/blob/master/schemas/SDMXStructureBase.xsd#L49

**Comment by stratosn** [comment]
Thanks! It looks like a bug.
cc @dosse

**Comment by dosse** [comment]
@achronis I agree that the parent should also be defined by the normal `IDType`. This is the case for [SDMX-JSON](https://github.com/sdmx-twg/sdmx-json/blob/71fe5eaa9fcd29e3c15f2f0216a19b9b650b1dbd/structure-message/tools/schemas/2.0.0/sdmx-json-structure-schema.json#L2505).

@jgager @glenntav Could you please confirm that the Code's Parent property should simply be of type `IDType`?

**Comment by jgager** [comment]
I believe this should be IDType. The rationale behind the NCNameIDType was to restrict component names so that elements and attributes generated in data structure specific schemas would be valid. I am not sure if there was ever a scenario where a code from a dimension code list would have been generated as an element/attribute in a data message (maybe the old cross sectional formats). But that may be why this was in there to begin with and just ended up being over looked. In any case, either code identifier or this parent code reference is the wrong type. I suspect the latter.

**Comment by dosse** [comment]
https://github.com/search?q=repo%3Asdmx-twg%2Fsdmx-ml%20parent&type=code

**Comment by dosse** [comment]
Ready to merge into Develop branch (branch patch-v3.0.1).

**Comment by stratosn** [comment]
@dosse @agent96 we are facing an issue with this, because the 3.0 XML schema is now invalid and kind of unusable. Should we foresee an update of the 3.0 schema in that context?

**Comment by agent96** [comment]
@stratosn @dosse There was a mechanism added when 3.1 was released for allowing bug fixes to be applied to the schema in a transparent way. On xml.sdmx.org the schemas are published, we can publish a 3.0.1 schema with the bug fix added, and then create a schema location of 3.0 (not third part to the version) which resolves to the latest 3.0.x schema.  We have this for the 3.1 schema already. The namespace of the schema remains the same, so tools can continue to work as normal.

**Comment by stratosn** [comment]
Thanks @agent96  - especially for XML it is important to maintain the namespace intact for any fixes.
I reopen the issue so we can fix it (unless you prefer to create a new ticket).

**Comment by dosse** [comment]
Hi @pedroacarranza, could I please ask you to follow this up?  

The same changes as done in 3.1:
- [Update SDMXStructureConcept.xsd](https://github.com/sdmx-twg/sdmx-ml/commit/4dc6a46a6f7b8b3fa3f49f6750f6a44d24b8fc16) [4dc6a46](https://github.com/sdmx-twg/sdmx-ml/commit/4dc6a46a6f7b8b3fa3f49f6750f6a44d24b8fc16)
- [Update SDMXStructureBase.xsd](https://github.com/sdmx-twg/sdmx-ml/commit/bce535e9e7a3844563c155f836c11ac6f761a930) [bce535e](https://github.com/sdmx-twg/sdmx-ml/commit/bce535e9e7a3844563c155f836c11ac6f761a930)
- [Update SDMXStructureCodelist.xsd](https://github.com/sdmx-twg/sdmx-ml/commit/769d75c1131f96d916608da1ec7799bdd5b52624) [769d75c](https://github.com/sdmx-twg/sdmx-ml/commit/769d75c1131f96d916608da1ec7799bdd5b52624)

should be done also in 3.0  
Thanks in advance.
